CotEditor/Sources/Script.swift

```diff
@@ -28,68 +28,163 @@
 
 import Cocoa
 
-class Script {
+enum ScriptingEventType: String {
     
-    let url: URL
-    let name: String
+    case documentOpened = "document opened"
+    case documentSaved = "document saved"
     
     
-    
-    // MARK: -
-    // MARK: Lifecycle
-    
-    init(url: URL, name: String) {
+    var eventID: AEEventID {
         
-        self.url = url
-        self.name = name
+        switch self {
+        case .documentOpened: return AEEventID(code: "edod")
+        case .documentSaved: return AEEventID(code: "edsd")
+        }
     }
     
+}
+
+
+
+enum ScriptingFileType {
+    
+    case appleScript
+    case shellScript
+    case unknown
```

Comment:
I feel `unknown` is not a good name. How about something like `folder` or `group. Or, just remove `unknown` and make `type` property in ScriptDescriptor optional.